src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -161,6 +161,21 @@ protected MultiBranchProject(ItemGroup parent, String name) {
     public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
         super.onLoad(parent, name);
         init2();
+        // migrate any non-mangled names
+        ArrayList<P> items = new ArrayList<>(getItems());
+        for (P item : items) {
+            String itemName = item.getName();
+            String mangledName = NameMangler.apply(itemName);
```

Comment:
Should this not rather be applying `NameMangler` to the original branch name? Here you are applying it to the encoded name, which seems like it might produce a different result.